### DISCUSSION TITLE
Can AssertJ filter elements for assertion?

### DISCUSSION THREAD

**Discussion by 1506085843** [assertj/assertj#3107] [Q&A] [answered]
`
    public class Student {
        public String name;

        public Student(String name) { this.name = name; }

        public String getName() {return name;}

        public void setName(String name) { this.name = name; }
    }

    @Test
    public void listFilterAssertThat() {
        List<Student> stuList = new ArrayList<>();
        stuList.add(new Student("Jerry" ));
        stuList.add(new Student("Tom"));
        stuList.add(new Student("Lucy"));

        //I want to check if the number of Tom is 1，but expected size: 1 but was: 3 in
        assertThat(stuList)
                .extracting(Student::getName)
                .asList()
                .contains("Tom")
                .hasSize(1);
    }
`

It's not working properly.Expected size: 1 but was: 3 in. I want to check if the number of Tom is 1 . I was wondering if there is a way to filter List elements and assert,like below：
`
assertEquals(1, stuList.stream() .filter(v -> "Tom".equals(v.getName())) .count() );
`

**Comment by joel-costigliola** [comment, marked as answer]
have you tried https://www.javadoc.io/doc/org.assertj/assertj-core/latest/org/assertj/core/api/AbstractIterableAssert.html#filteredOn(java.util.function.Predicate)?
